src/bun.js/node/node_fs_watcher.zig

```diff
@@ -403,6 +404,32 @@ pub const FSWatcher = struct {
 
                             current_task.append(relative_path, event_type, .destroy);
 
+                            const full_path = bun.default_allocator.dupeZ(u8, path_slice) catch unreachable;
+                            const is_watched = blk: {
+                                for (this.file_paths.slice()) |path| {
+                                    if (std.mem.eql(u8, path, full_path)) {
+                                        break :blk true;
+                                    }
+                                }
+                                break :blk false;
+                            };
+
+                            // Add new files or directories to the watch list.
+                            if (!is_watched) {
+                                this.file_paths.push(bun.default_allocator, full_path) catch unreachable;
+                                const fs_info = fdFromAbsolutePathZ(full_path) catch null;
```

Comment:
We definitely don't have an existing file descriptor for this file path, right? This is a very expensive system call so we should be sure to only do it if definitely necessary